rnaseq-cgl-pipeline/wrapper.py

```diff
@@ -0,0 +1,96 @@
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger()
+
+
+def call_pipeline(mount, args):
+    uuid = 'Toil-RNAseq-' + str(uuid4())
+    if not os.path.exists(mount + uuid):
+        os.makedirs(os.path.join(mount, uuid))
+    os.environ['PYTHONPATH'] = '/opt/rnaseq-pipeline/src'
+    command = ['python', '-m', 'toil_scripts.rnaseq_cgl.rnaseq_cgl_pipeline',
+               os.path.join(mount, 'jobStore'),
+               '--retryCount', '1',
+               '--output-dir', mount,
+               '--workDir', os.path.join(mount, uuid),
+               '--star-index', 'file://'+ args.star,
+               '--rsem-ref', 'file://' + args.rsem,
+               '--kallisto-index', 'file://' + args.kallisto,
+               '--sample-urls']
+    command.extend(['file://' + x for x in args.samples])
+    if args.restart:
+        command.append('--restart')
+    try:
+        subprocess.check_call(command)
+    finally:
+        user_id = subprocess.check_output(['stat', '-c', '%u:%g', mount]).strip()
+        subprocess.check_call(['chown', '-R', user_id, mount])
+        shutil.rmtree(os.path.join(mount, uuid))
+
+
+def main():
+    """
+    Please see the complete documentation located at:
+    https://github.com/BD2KGenomics/cgl-docker-lib/tree/master/rnaseq-cgl-pipeline
+    or in the container at:
+    /opt/rnaseq-pipeline/README.md
+
+    All samples and inputs must be reachable via Docker "-v" mount points and use
+    the Destination path prefix.
+    """
+    # Define argument parser for
+    parser = argparse.ArgumentParser(description=main.__doc__, formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument('--star', type=str, required=True,
+                        help='Absolute path to Star Index tarball.')
+    parser.add_argument('--rsem', type=str, required=True,
+                        help='Absolute path to rsem reference tarball.')
+    parser.add_argument('--kallisto', type=str, required=True,
+                        help='Absolute path to kallisto index (.idx) file.')
+    parser.add_argument('--samples', nargs='+', required=True,
+                        help='Absolute path(s) to sample tarballs.')
+    parser.add_argument('--restart', action='store_true', default=False,
+                        help='Add this flag to restart the pipeline. Requires existing job store.')
+    args = parser.parse_args()
+    # Get name of most recent running container (should be this one)
+    name = subprocess.check_output(['docker', 'ps', '--format', '{{.Names}}']).split('\n')[0]
+    # Get name of mounted volume
+    blob = json.loads(subprocess.check_output(['docker', 'inspect', name]))
+    mounts = blob[0]['Mounts']
+    # Ensure docker.sock is mounted correctly
+    sock_mount = [x['Source'] == x['Destination'] for x in mounts if 'docker.sock' in x['Source']]
+    if len(sock_mount) != 1:
+        raise IllegalArgumentException('Missing socket mount: docker run -v /var/run/docker.sock:/var/run/docker.sock')
```

Comment:
It's a bit unclear if the part after the colon is the problem or the resolution.